packages/graphql-fixtures/src/fill.ts

```diff
@@ -198,8 +201,23 @@ function unwrapThunk<T>(value: Thunk<T>, details: ResolveDetails): T {
   return isResolver(value) ? value({...details, type: unwrappedType}) : value;
 }
 
-function withRandom<T>(keypath: FieldDetails[], func: () => T) {
-  faker.seed(seedFromKeypath(keypath.map(({responseName}) => responseName)));
+function keyPathElement(responseName: string, fieldIndex: number | undefined) {
+  return fieldIndex == null ? responseName : `${responseName}[${fieldIndex}]`;
+}
+
+function withRandom<T>(
+  // {fieldIndex = 0}: FieldDetails,
+  keypath: FieldDetails[],
+  func: () => T,
+  fieldIndex = 0,
+) {
+  faker.seed(
+    seedFromKeypath(
+      keypath.map(({fieldIndex, responseName}) =>
+        keyPathElement(responseName, fieldIndex),
+      ),
+    ) + fieldIndex,
```

Comment:
Is this still needed?

Reply:
i believe so. there should be two cases, one where one of our ancestors is indexed and another where the field is indexed as a leaf in the graph. This `+ fieldIndex` covers the leaf scenario. The alternative is to inject the field being filled into the keypath, but this just seemed a bit easier at first glance. the other method (essentially `withRandom([...details.parentFields, details])`) is probably just as viable but slightly less performant.